manim/mobject/svg/svg_mobject.py

```diff
@@ -559,93 +562,140 @@ def generate_points(self):
 
-        if command == "C":  # curveto
-            pass  # Yay! No action required
-        elif command in ["S", "T"]:  # smooth curveto
-            self.add_smooth_curve_to(*new_points)
-            # handle1 = points[-1] + (points[-1] - points[-2])
-            # new_points = np.append([handle1], new_points, axis=0)
+        elif command == "S":  # Smooth cubic
+            prev_handle = start_point
+            if prev_command in ["C", "S"]:
+                prev_handle = self.points[-2]
+            for i in range(0, len(new_points), 2):
+                new_handle = 2 * start_point - prev_handle
+                self.add_cubic_bezier_curve_to(new_handle, new_points[i], new_points[i+1])
+                start_point = new_points[i+1]
+                prev_handle = new_points[i]
             return
+
         elif command == "Q":  # quadratic Bezier curve
-            # TODO, this is a suboptimal approximation
-            new_points = np.append([new_points[0]], new_points, axis=0)
+            for i in range(0, len(new_points), 2):
+                self.add_quadratic_bezier_curve_to(new_points[i], new_points[i+1])
+            return
+
+        elif command == "T":  # smooth quadratic
+            prev_quad_handle = start_point
+            if prev_command in ["Q", "T"]:
+                # because of the conversion from quadratic to cubic,
+                # our actual previous handle was 3/2 in the direction of p[-2] from p[-1]
+                prev_quad_handle = 1.5 * self.points[-2] - 0.5 * self.points[-1]
+            for p in new_points:
+                new_quad_handle = 2 * start_point - prev_quad_handle
+                self.add_quadratic_bezier_curve_to(new_quad_handle, p)
+                start_point = p
+                prev_quad_handle = new_quad_handle
+
         elif command == "A":  # elliptical Arc
             raise NotImplementedError()
+
         elif command == "Z":  # closepath
+            self.add_line_to(self.current_path_start)
             return
 
-        # Add first three points
-        self.add_cubic_bezier_curve_to(*new_points[0:3])
-
-        # Handle situations where there's multiple relative control points
-        if len(new_points) > 3:
-            # Add subsequent offset points relatively.
-            for i in range(3, len(new_points), 3):
-                if isLower:
-                    new_points[i : i + 3] -= points[-1]
-                    new_points[i : i + 3] += new_points[i - 1]
-                self.add_cubic_bezier_curve_to(*new_points[i : i + 3])
-
-    def string_to_points(self, coord_string):
+    def string_to_points(self, command, is_relative, coord_string, start_point):
         """Since the SVG file's path command is provided as a string, this
-        converts the coordinates into numbers.
+        converts the coordinates into numbers. If the command is lower,
+        it also converts the values from values relative to values relative to the previous final point
         """
```

Comment:
Further into the development of this PR, I would like to see full documentation for each of these methods, including a one line summary, an extended summary, explanation of each of the parameters, etc. Currently, this docstring reads more like a comment than documentation.